File: general/javascript/codility/4.1 Frog-River-One.cc
int solution(int X, vector<int> &A){
    vector<int> B(X, -1);
    for(int i=0; i<A.size(); i++){
        if(A[i] <=X && B[A[i] - 1]=== -1) B[A[i] - 1] = i;
    }

    int max = 0;
    for (int i=0; i<B.size(); i++){
        if(B[i]== -1) return -1;

        if(max < B[i]) max=B[i];
    }

    return -1;
}

File: general/javascript/codility/4.1 Frog-River-One.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(X, A) {
    // Implement your solution here
    let B = Array(X).fill(-1);

    for(let i=0; i<A.length; i++){
        if(A[i] <= X && B[A[i] - 1]== -1){
            B[A[i] - 1] = i;
        }
    }
    // if(B.includes(-1)) return -1;
    // console.log(B)
    let max = 0;
    for(let i=0; i<B.length; i++){
        if(B[i] == -1) return -1;
        if(B[i] > max){
            max = B[i]
        }
    }

    return max;
}
